File: src/sttp/sttp-core-rs/src/application/services/rekey_scope_service.rs
use std::sync::Arc;

use anyhow::Result;

use crate::domain::contracts::NodeStore;
use crate::domain::models::BatchRekeyResult;

pub struct RekeyScopeService {
    store: Arc<dyn NodeStore>,
}

impl RekeyScopeService {
    pub fn new(store: Arc<dyn NodeStore>) -> Self {
        Self { store }
    }

    pub async fn rekey_async(
        &self,
        node_ids: Vec<String>,
        target_tenant_id: &str,
        target_session_id: &str,
        dry_run: bool,
        allow_merge: bool,
    ) -> Result<BatchRekeyResult> {
        self.store
            .batch_rekey_scopes_async(
                node_ids,
                target_tenant_id,
                target_session_id,
                dry_run,
                allow_merge,
            )
            .await
    }
}

File: src/sttp/sttp-core/Application/Services/RekeyScopeService.cs
using SttpMcp.Domain.Contracts;
using SttpMcp.Domain.Models;

namespace SttpMcp.Application.Services;

public sealed class RekeyScopeService(INodeStore store)
{
    public Task<BatchRekeyResult> RekeyAsync(
        IReadOnlyList<string> nodeIds,
        string targetTenantId,
        string targetSessionId,
        bool dryRun,
        bool allowMerge,
        CancellationToken ct = default)
        => store.BatchRekeyScopesAsync(nodeIds, targetTenantId, targetSessionId, dryRun, allowMerge, ct);
}